Index: linux-5.4.211/drivers/net/dsa/mt7530.c
===================================================================
--- linux-5.4.211.orig/drivers/net/dsa/mt7530.c
+++ linux-5.4.211/drivers/net/dsa/mt7530.c
@@ -1000,8 +1000,11 @@ mt7530_port_enable(struct dsa_switch *ds
 	 */
 	priv->ports[port].pm |= PCR_MATRIX(BIT(MT7530_CPU_PORT));
 	priv->ports[port].enable = true;
-	mt7530_rmw(priv, MT7530_PCR_P(port), PCR_MATRIX_MASK,
-		   priv->ports[port].pm);
+	/* Disable isolation port matrix setting here, this setting will isloate
+	 * the port traffic in switch when each time port/dsa interface down/up.
+	 */
+	//mt7530_rmw(priv, MT7530_PCR_P(port), PCR_MATRIX_MASK,
+	//	   priv->ports[port].pm);
 	mt7530_clear(priv, MT7530_PMCR_P(port), PMCR_LINK_SETTINGS_MASK);
 
 	mutex_unlock(&priv->reg_mutex);
@@ -1023,8 +1026,11 @@ mt7530_port_disable(struct dsa_switch *d
 	 * enablement for the port.
 	 */
 	priv->ports[port].enable = false;
-	mt7530_rmw(priv, MT7530_PCR_P(port), PCR_MATRIX_MASK,
-		   PCR_MATRIX_CLR);
+	/* Disable port matrix clear. Port/dsa interface down/up will invoked
+	 * this function.
+	 */
+	//mt7530_rmw(priv, MT7530_PCR_P(port), PCR_MATRIX_MASK,
+	//	   PCR_MATRIX_CLR);
 	mt7530_clear(priv, MT7530_PMCR_P(port), PMCR_LINK_SETTINGS_MASK);
 
 	mutex_unlock(&priv->reg_mutex);
@@ -1840,6 +1846,13 @@ mt7531_setup(struct dsa_switch *ds)
 		/* Enable consistent egress tag */
 		mt7530_rmw(priv, MT7530_PVC_P(i), PVC_EG_TAG_MASK,
 			   PVC_EG_TAG(MT7530_VLAN_EG_CONSISTENT));
+
+		/* Set port matrix on all user ports to support BOOTUP_NOLOOP
+		 * function and multiboot service.
+		 */
+		if (!dsa_is_cpu_port(ds, i))
+			mt7530_rmw(priv, MT7530_PCR_P(i), PCR_MATRIX_MASK,
+				PCR_MATRIX(BIT(i) + BIT(MT7530_CPU_PORT)));
 	}
 
 	mt7531_phy_setup(ds);
