--- a/include/uapi/linux/wireless.h
+++ b/include/uapi/linux/wireless.h
@@ -428,7 +428,7 @@
 /* ----------------------- OTHER CONSTANTS ----------------------- */
 
 /* Maximum frequencies in the range struct */
-#define IW_MAX_FREQUENCIES	32
+#define IW_MAX_FREQUENCIES	59
 /* Note : if you have something like 80 frequencies,
  * don't increase this constant and don't fill the frequency list.
  * The user will be able to set by channel anyway... */
@@ -446,7 +446,7 @@
 
 /* Maximum of address that you may get in the
    list of access points in range */
-#define IW_MAX_AP		64
+#define IW_MAX_AP		128
 
 /* Maximum size of the ESSID and NICKN strings */
 #define IW_ESSID_MAX_SIZE	32
