--- a/scripts/netifd-wireless.sh
+++ b/scripts/netifd-wireless.sh
@@ -31,7 +31,8 @@ prepare_key_wep() {
 
 	echo -n "$key" | grep -qE "[^a-fA-F0-9]" && hex=0
 	[ "${#key}" -eq 10 -a $hex -eq 1 ] || \
-	[ "${#key}" -eq 26 -a $hex -eq 1 ] || {
+	[ "${#key}" -eq 26 -a $hex -eq 1 ] || \
+	[ "${#key}" -eq 32 -a $hex -eq 1 ] || {
 		[ "${key:0:2}" = "s:" ] && key="${key#s:}"
 		key="$(echo -n "$key" | hexdump -ve '1/1 "%02x" ""')"
 	}
@@ -108,14 +109,16 @@ _wdev_wrapper() {
 }
 
 _wdev_notify_init() {
-	local command="$1"
-	local name="$2"
-	local value="$3"
+	local command="$1"; shift;
 
 	json_init
 	json_add_int "command" "$command"
 	json_add_string "device" "$__netifd_device"
-	[ -n "$name" -a -n "$value" ] && json_add_string "$name" "$value"
+	while [ -n "$1" ]; do
+		local name="$1"; shift
+		local value="$1"; shift
+		json_add_string "$name" "$value"
+	done
 	json_add_object "data"
 }
 
@@ -151,7 +154,7 @@ _wireless_add_vlan() {
 	local name="$1"; shift
 	local ifname="$1"; shift
 
-	_wdev_notify_init $CMD_SET_DATA "vlan" "$name"
+	_wdev_notify_init $CMD_SET_DATA interface "$__cur_interface" "vlan" "$name"
 	json_add_string "ifname" "$ifname"
 	_wdev_add_variables "$@"
 	_wdev_notify
@@ -230,6 +232,7 @@ case "$encryption" in
 		*tkip) wpa_cipher="TKIP";;
 		*gcmp256) wpa_cipher="GCMP-256";;
 		*gcmp) wpa_cipher="GCMP";;
+		wpa3-192*) wpa_cipher="GCMP-256";;
 	esac
 
 	# 802.11n requires CCMP for WPA
@@ -261,21 +265,27 @@ case "$encryption" in
 		owe*)
 			auth_type=owe
 		;;
+		wpa3-192*)
+			auth_type=eap192
+		;;
 		wpa3-mixed*)
-			auth_type=eap-eap192
+			auth_type=eap2-eap3
 		;;
 		wpa3*)
-			auth_type=eap192
+			auth_type=eap3
 		;;
 		psk3-mixed*|sae-mixed*)
 			auth_type=psk-sae
 		;;
 		psk3*|sae*)
 			auth_type=sae
 		;;
 		*psk*)
 			auth_type=psk
 		;;
+		*wpa2*)
+			auth_type=eap2
+		;;
 		*wpa*|*8021x*)
 			auth_type=eap
 		;;
@@ -336,6 +346,7 @@ for_each_interface() {
 				continue
 			}
 		fi
+		__cur_interface="$_w_iface"
 		"$@" "$_w_iface"
 		json_select ..
 	done
@@ -378,10 +389,12 @@ for_each_interface() {
 
 _wdev_common_iface_config() {
 	config_add_string mode ssid encryption 'key:wpakey'
+	config_add_boolean bridge_isolate
 }
 
 _wdev_common_vlan_config() {
 	config_add_string name vid iface
+	config_add_boolean bridge_isolate
 }
 
 _wdev_common_station_config() {
