--- a/mt_wifi/embedded/common/multi_profile.c
+++ b/mt_wifi/embedded/common/multi_profile.c
@@ -1767,9 +1767,9 @@ static INT multi_profile_merge_mbss(
 	/*merge security*/
 	multi_profile_merge_security(ad, data, buf1, buf2, final);
 	/*merge WmmCapable*/
-	multi_profile_merge_separate("WmmCapable", buf1, buf2, final);
+	multi_profile_merge_perbss(data, "WmmCapable", buf1, buf2, final, MPF_APPEND_BSS0);
 	/*merge NoForwarding*/
-	multi_profile_merge_separate("NoForwarding", buf1, buf2, final);
+	multi_profile_merge_perbss(data, "NoForwarding", buf1, buf2, final, MPF_APPEND_BSS0);
 	/*merge StationKeepAlive*/
 	multi_profile_merge_perbss(data, "StationKeepAlive", buf1, buf2, final, MPF_APPEND_BSS0);
 	/*merge HideSSID*/
@@ -3628,6 +3628,21 @@ #ifdef TXBF_SUPPORT
 
 	return NDIS_STATUS_SUCCESS;
 }
+#ifdef MBSS_AS_WDS_AP_SUPPORT
+static INT multi_profile_merge_mbss_wds(
+	struct mpf_data *data,
+	CHAR *buf1,
+	CHAR *buf2,
+	CHAR *final)
+{
+	/* merge WDSEnable*/
+	multi_profile_merge_perbss(data, "WDSEnable", buf1, buf2, final, MPF_APPEND_BSS0);
+	/* merge WdsMac */
+	multi_profile_merge_perbss(data, "WdsMac", buf1, buf2, final, MPF_APPEND_BSS0);
+
+	return NDIS_STATUS_SUCCESS;
+}
+#endif /* MBSS_AS_WDS_AP_SUPPORT */
 #endif /* DOT11_HE_AX */
 #endif /* TXBF_SUPPORT */
 #endif /* n MBSS_SUPPORT */
@@ -3649,12 +3664,13 @@
 
 #ifdef MWDS
 static INT multi_profile_merge_mwds(
+	struct mpf_data *data,
 	CHAR *buf1,
 	CHAR *buf2,
 	CHAR *final)
 {
 	/*merge ApMWDS*/
-	multi_profile_merge_separate("ApMWDS", buf1, buf2, final);
+	multi_profile_merge_perbss(data, "ApMWDS", buf1, buf2, final, MPF_APPEND_BSS0);
 	/*merge ApCliMWDS*/
 	multi_profile_merge_separate("ApCliMWDS", buf1, buf2, final);
 	return NDIS_STATUS_SUCCESS;
@@ -3747,6 +3763,14 @@ static INT multi_profile_merge_ft(
 {
 	/*merge FtMdId*/
 	multi_profile_merge_increase(data, 1, "FtMdId", buf1, buf2, final);
+	/*merge FtR0khId*/
+	multi_profile_merge_increase(data, 1, "FtR0khId", buf1, buf2, final);
+	/*merge FtR1khId*/
+	multi_profile_merge_increase(data, 1, "FtR1khId", buf1, buf2, final);
+	/*merge AssocDeadLine*/
+	multi_profile_merge_increase(data, 1, "AssocDeadLine", buf1, buf2, final);
+	/*merge R0KeyLifeTime*/
+	multi_profile_merge_increase(data, 1, "R0KeyLifeTime", buf1, buf2, final);
 	/*merge FtSupport */
 	multi_profile_merge_perbss(data, "FtSupport", buf1, buf2, final, MPF_APPEND_BSS0);
 	/*merge FtOnly */
@@ -3766,12 +3790,27 @@ static INT multi_profile_merge_rrm(
 	CHAR *buf2,
 	CHAR *final)
 {
-	/*merge FtSupport */
+	/*merge RRMEnable */
-	multi_profile_merge_separate("RRMEnable", buf1, buf2, final);
+	multi_profile_merge_perbss(data, "RRMEnable", buf1, buf2, final, MPF_APPEND_BSS0);
 	return NDIS_STATUS_SUCCESS;
 }
 #endif
 
+#ifdef CONFIG_DOT11V_WNM
+static INT multi_profile_merge_wnm(
+	struct mpf_data *data,
+	CHAR *buf1,
+	CHAR *buf2,
+	CHAR *final)
+{
+	/*merge WNMEnable */
+	multi_profile_merge_perbss(data, "WNMEnable", buf1, buf2, final, MPF_APPEND_BSS0);
+	multi_profile_merge_perbss(data, "ProxyARPEnable", buf1, buf2, final, MPF_APPEND_BSS0);
+	multi_profile_merge_perbss(data, "WNMNotifyEnable", buf1, buf2, final, MPF_APPEND_BSS0);
+	return NDIS_STATUS_SUCCESS;
+}
+#endif
+
 #ifdef DSCP_PRI_SUPPORT
 INT multi_profile_merge_dscp_pri(
 	struct mpf_data *data,
@@ -4086,6 +4125,11 @@
 	if (multi_profile_merge_ersu(buf1, buf2, final) != NDIS_STATUS_SUCCESS)
 		return retval;
 #endif /* DOT11_HE_AX */
+
+#ifdef MBSS_AS_WDS_AP_SUPPORT
+	if (multi_profile_merge_mbss_wds(data, buf1, buf2, final) != NDIS_STATUS_SUCCESS)
+		return retval;
+#endif /* MBSS_AS_WDS_AP_SUPPORT */
 
 #ifdef DYNAMIC_WMM_SUPPORT
 	if multi_profile_merge_separate("DynWmmEnable", buf1, buf2, final) != NDIS_STATUS_SUCCESS)
@@ -4110,7 +4154,7 @@
 #endif /* LINK_TEST_SUPPORT */
 
 #ifdef MWDS
-	if (multi_profile_merge_mwds(buf1, buf2, final) != NDIS_STATUS_SUCCESS)
+	if (multi_profile_merge_mwds(data, buf1, buf2, final) != NDIS_STATUS_SUCCESS)
 		return retval;
 #endif
 
@@ -4147,6 +4191,11 @@ #ifdef DOT11K_RRM_SUPPORT
 		return retval;
 #endif
 
+#ifdef CONFIG_DOT11V_WNM
+	if (multi_profile_merge_wnm(data, buf1, buf2, final) != NDIS_STATUS_SUCCESS)
+		return retval;
+#endif
+
 #ifdef DSCP_PRI_SUPPORT
 	if (multi_profile_merge_dscp_pri(data, buf1, buf2, final) != NDIS_STATUS_SUCCESS)
 		return retval;
