--- a/mt_wifi_ap/Kconfig
+++ b/mt_wifi_ap/Kconfig
@@ -287,7 +287,6 @@ #WDS STA
 config WDS_STA_SUPPORT
 	bool "WDS STA Support"
 	depends on APCLI_SUPPORT
-	depends on CFG80211_SUPPORT
         depends on ENTERPRISE_AP_SUPPORT
 	default n
 
--- a/mt_wifi/os/linux/Kconfig.mt_wifi_ap
+++ b/mt_wifi/os/linux/Kconfig.mt_wifi_ap
@@ -287,7 +287,6 @@ #WDS STA
 config WDS_STA_SUPPORT
 	bool "WDS STA Support"
 	depends on APCLI_SUPPORT
-	depends on CFG80211_SUPPORT
         depends on ENTERPRISE_AP_SUPPORT
 	default n
 
--- a/mt_wifi/embedded/common/capi.c
+++ b/mt_wifi/embedded/common/capi.c
@@ -746,7 +746,7 @@ switch (mode) {
 					"Set %s by wcid success.\n", gi_ltf_info[gi]);
 			}
 			else
-				MTWF_DBG(pAd, DBG_CAT_TEST, CATTEST_RFEATURE, DBG_LVL_INFO, "Invalid wcid value(%d)!!!.\n", wcid));
+				MTWF_DBG(pAd, DBG_CAT_TEST, CATTEST_RFEATURE, DBG_LVL_INFO, "Invalid wcid value(%d)!!!.\n", wcid);
 			break;
 		case GI_BY_BSS:
 			if (!wdev) {
--- a/mt_wifi/embedded/include/rtmp.h
+++ b/mt_wifi/embedded/include/rtmp.h
@@ -3613,25 +3613,25 @@ typedef struct _COMMON_CONFIG {
 #ifdef HOSTAPD_MAP_SUPPORT
 	BOOLEAN bHostapdMapDisabled;
 #endif
-#ifdef EAP_STATS_SUPPORT
-	BOOLEAN bEapStatsDisabled;
-#endif
 #ifdef RADIUS_MAC_AUTH_SUPPORT
 	BOOLEAN bRadiusMacAuthDisabled;
 #endif
 #ifdef DYNAMIC_VLAN_SUPPORT
 	BOOLEAN bDynamicVlanDisabled;
 #endif
+#endif
+#ifdef EAP_STATS_SUPPORT
+	BOOLEAN bEapStatsDisabled;
+#endif
 #ifdef MBSS_AS_WDS_AP_SUPPORT
 	BOOLEAN bMBSSASWDSAPDisabled;
 #endif
 #ifdef CLIENT_WDS
 	BOOLEAN bClientWdsDisabled;
 #endif
 #ifdef APCLI_AS_WDS_STA_SUPPORT
 	BOOLEAN bApcliASWDSSTADisabled;
 #endif
-#endif
 #ifdef ANDLINK_FEATURE_SUPPORT
 	UINT8 andlink_enable[DBDC_BAND_NUM];
 	UINT8 andlink_ip_hostname_en[DBDC_BAND_NUM];
--- a/mt_wifi/embedded/mcu/andes_mt.c
+++ b/mt_wifi/embedded/mcu/andes_mt.c
@@ -2997,11 +2997,10 @@ static VOID ExtEventGetAllStaStats(RTMP_
 #endif
 	}
 	break;
+
 	case EVENT_PHY_PER_STA_TX_STAT:
-#ifdef RT_CFG80211_SUPPORT
+#ifdef EAP_STATS_SUPPORT
 	if (!pAd->CommonCfg.bEapStatsDisabled) {
-#endif
-#ifdef EAP_STATS_SUPPORT
 		P_EXT_EVENT_TX_STAT_RESULT_T prEventExtCmdResult = (P_EXT_EVENT_TX_STAT_RESULT_T)Data;
 
 		prEventExtCmdResult->u2StaNum = le2cpu16(prEventExtCmdResult->u2StaNum);
@@ -3022,19 +3021,15 @@ static VOID ExtEventGetAllStaStats(RTMP_
 			}
 
 		}
-#endif
-#ifdef RT_CFG80211_SUPPORT
 	}
 #endif
 	break;
 
 	case EVENT_PHY_RX_STAT:
-#ifdef RT_CFG80211_SUPPORT
+#ifdef EAP_STATS_SUPPORT
 	if (!pAd->CommonCfg.bEapStatsDisabled) {
-#endif
-#ifdef EAP_STATS_SUPPORT
 		P_EXT_EVENT_RX_STAT_RESULT_T prEventExtCmdResult = (P_EXT_EVENT_RX_STAT_RESULT_T)Data;
-		UCHAR       concurrent_bands = HcGetAmountOfBand(pAd);
+		UCHAR concurrent_bands = HcGetAmountOfBand(pAd);
 
 		for (Idx = 0; Idx < concurrent_bands; Idx++) {
 			P_EXT_EVENT_RX_STAT_T rRxStatResult = &prEventExtCmdResult->rRxStatResult[Idx];
@@ -3044,10 +3039,8 @@ static VOID ExtEventGetAllStaStats(RTMP_
 			pAd->OneSecMibBucket.MdrdyCount[Idx] = le2cpu16(rRxStatResult->u2PhyRxMdrdyCntCck) +
 													le2cpu16(rRxStatResult->u2PhyRxMdrdyCntOfdm);
 		}
-#endif
-#ifdef RT_CFG80211_SUPPORT
 	}
 #endif
 	break;
 
 	case EVENT_PHY_PER_STA_TXRX_AIR_TIME:
--- a/mt_wifi/embedded/sta/sta.c
+++ b/mt_wifi/embedded/sta/sta.c
@@ -1638,8 +1638,10 @@ BOOLEAN apcli_fill_non_offload_tx_blk(RTMP_ADAPTER
 #ifdef APCLI_AS_WDS_STA_SUPPORT
 					if (pAd->StaCfg[pMacEntry->func_tb_idx].wdev.wds_enable == 0 || pAd->CommonCfg.bApcliASWDSSTADisabled)
 #endif /* APCLI_AS_WDS_STA_SUPPORT */
+					{
 						apCliPkt = (PNDIS_PACKET)MATEngineTxHandle(pAd, pPacket, pMacEntry->func_tb_idx, pMacEntry->EntryType);
 						pMacAddr = &pAd->StaCfg[pMacEntry->func_tb_idx].wdev.if_addr[0];
+					}
 					}
 
 					if (apCliPkt) {
