--- a/mt_wifi/embedded/include/rtmp.h
+++ b/mt_wifi/embedded/include/rtmp.h
@@ -5312,9 +5312,10 @@ typedef enum _MAC_ENT_STATUS_ {
 } MAC_ENT_STATUS;
 
 #define BAND_NUM_MAX 2
+#define MAX_NUMBER_OF_MAC 544
 typedef struct _MAC_TABLE {
 	MAC_TABLE_ENTRY * Hash[HASH_TABLE_SIZE];
-	MAC_TABLE_ENTRY Content[MAX_LEN_OF_MAC_TABLE];
+	MAC_TABLE_ENTRY Content[MAX_NUMBER_OF_MAC];
 	STA_TR_ENTRY tr_entry[MAX_LEN_OF_TR_TABLE];
 	/*
 		Be care in mgmt_entrytb.c  MacTableReset() will NdisZeroMemory(&pAd->MacTab.Size, sizeof(MAC_TABLE)-offsetof(MAC_TABLE, Size));
--- a/mt_wifi/embedded/common/cmm_info.c
+++ b/mt_wifi/embedded/common/cmm_info.c
@@ -7256,6 +7256,11 @@ typedef struct _RT_802_11_MAC_ENTRY_FIX
 	HTTRANSMIT_SETTING_FIX LastRxRate;
 	SHORT StreamSnr[3];				/* BF SNR from RXWI. Units=0.25 dB. 22 dB offset removed */
 	SHORT SoundingRespSnr[3];			/* SNR from Sounding Response. Units=0.25 dB. 22 dB offset removed */
+	UINT64 TxPackets;  //TxPackets.QuadPart
+	UINT64 RxPackets;  //RxPackets.QuadPart
+	UINT64 TxBytes;
+	UINT64 RxBytes;
+	UINT32 ExPected_throughput;
 	UINT32 EncryMode;
 	UINT32 AuthMode;
 } RT_802_11_MAC_ENTRY_FIX, *PRT_802_11_MAC_ENTRY_FIX;
@@ -7380,9 +7385,14 @@ VOID RTMPIoctlGetMacTableStaInfo(
 			pDst->LastRxRate.word = RTMPGetLastRxRate(pAd, pEntry);
 			pDst->EncryMode = pEntry->SecConfig.PairwiseCipher;
 			pDst->AuthMode = pEntry->SecConfig.AKMMap;
+			pDst->TxPackets = (UINT64)pEntry->TxPackets.QuadPart;
+			pDst->RxPackets = (UINT64)pEntry->RxPackets.QuadPart;
+			pDst->TxBytes = (UINT64)pEntry->TxBytes;
+			pDst->RxBytes = (UINT64)pEntry->RxBytes;
+			pDst->ExPected_throughput = pEntry->OneSecTxBytes; //pEntry->AvgRxBytes;
 			pMacTab->Num += 1;
 			/* Add to avoid Array cross board */
-			if (pMacTab->Num >= MAX_LEN_OF_MAC_TABLE)
+			if (pMacTab->Num >= MAX_NUMBER_OF_MAC) //MAX_NUMBER_OF_MAC
 				break;
 
 		}
