--- a/mt_wifi/embedded/wapp/wapp.c
+++ b/mt_wifi/embedded/wapp/wapp.c
@@ -541,6 +541,12 @@
 		}
 	}
 	cli_info->uplink_rssi = RTMPAvgRssi(pAd, &mac_entry->RssiSample);
+	
+	cli_info->uplink += 400;
+	cli_info->downlink += 400;
+	cli_info->uplink_rssi += 30;
+	if (WMODE_CAP_5G(mac_entry->wdev->PhyMode))
+	{cli_info->uplink_rssi -=25;};
 #ifdef CONFIG_DOT11V_WNM
 	cli_info->cli_caps.btm_capable = mac_entry->bBSSMantSTASupport == TRUE ? 1 : 0;
 #endif
