--- a/iwpriv.c
+++ b/iwpriv.c
@@ -622,7 +622,7 @@ print_priv_info(int		skfd,
       /* Print them all */
       for(k = 0; k < n; k++)
 	if(priv[k].name[0] != '\0')
-	  printf("          %-16.16s (%.4X) : set %3d %s & get %3d %s\n",
+	  printf("          %-20.20s (%.4X) : set %3d %s & get %3d %s\n",
 		 priv[k].name, priv[k].cmd,
 		 priv[k].set_args & IW_PRIV_SIZE_MASK,
 		 argtype[(priv[k].set_args & IW_PRIV_TYPE_MASK) >> 12],
--- a/wireless.21.h
+++ b/wireless.21.h
@@ -1079,7 +1079,7 @@ struct	iw_priv_args
 	__u32		cmd;		/* Number of the ioctl to issue */
 	__u16		set_args;	/* Type and number of args */
 	__u16		get_args;	/* Type and number of args */
-	char		name[IFNAMSIZ];	/* Name of the extension */
+	char		name[20];	/* Name of the extension */
 };
 
 /* ----------------------- WIRELESS EVENTS ----------------------- */
