--- a/mt_wifi/embedded/ap/ap_autoChSel.c
+++ b/mt_wifi/embedded/ap/ap_autoChSel.c
@@ -67,10 +67,12 @@ static inline INT Get56GBandChOffset(
 #ifdef A_BAND_SUPPORT
 
 		if ((Channel == 36) || (Channel == 44) || (Channel == 52) || (Channel == 60) || (Channel == 100) || (Channel == 108) ||
-			(Channel == 116) || (Channel == 124) || (Channel == 132) || (Channel == 149) || (Channel == 157))
+			(Channel == 116) || (Channel == 124) || (Channel == 132) || (Channel == 140) || (Channel == 149) || (Channel == 157) ||
+			(Channel == 165) || (Channel == 173))
 			return 1;
 		else if ((Channel == 40) || (Channel == 48) || (Channel == 56) || (Channel == 64) || (Channel == 104) || (Channel == 112) ||
-				 (Channel == 120) || (Channel == 128) || (Channel == 136) || (Channel == 153) || (Channel == 161))
+				 (Channel == 120) || (Channel == 128) || (Channel == 136) || (Channel == 144) || (Channel == 153) || (Channel == 161) ||
+				 (Channel == 169) || (Channel == 177))
 			return -1;
 
 #endif /* A_BAND_SUPPORT */
@@ -2817,7 +2819,7 @@
 			pACSChList[ChIdx].CentralChannel = pACSChList[ChIdx].Channel;
 			pACSChList[ChIdx].BuildDone = TRUE;
 		} else if ((cfg_vht_bw == VHT_BW_160) && (vht_ch_160M != NULL)
-				   && vht80_channel_group(pAd, pACSChList[ChIdx].Channel, pwdev)) {
+				   && vht160_channel_group(pAd, pACSChList[ChIdx].Channel, pwdev)) {
 			idx = 0;
 			count = 0;
 
